extensions/stats/plugin.cc

```diff
@@ -537,51 +537,39 @@ void PluginRootContext::onTick() {
 
 bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
                                bool is_tcp) {
-  std::string peer_id;
-  bool peer_found = getValue({peer_metadata_id_key_}, &peer_id);
-
-  std::string peer;
-  const ::Wasm::Common::FlatNode* peer_node =
-      peer_found && getValue({peer_metadata_key_}, &peer)
-          ? flatbuffers::GetRoot<::Wasm::Common::FlatNode>(peer.data())
-          : nullptr;
+  Wasm::Common::PeerNodeInfo peer_node_info(peer_metadata_id_key_,
+                                            peer_metadata_key_);
 
   // map and overwrite previous mapping.
-  const ::Wasm::Common::FlatNode* destination_node_info =
-      outbound_ ? peer_node
-                : flatbuffers::GetRoot<::Wasm::Common::FlatNode>(
+  const ::Wasm::Common::FlatNode& destination_node_info =
+      outbound_ ? peer_node_info.get()
+                : *flatbuffers::GetRoot<::Wasm::Common::FlatNode>(
                       local_node_info_.data());
   std::string destination_namespace =
-      destination_node_info && destination_node_info->namespace_()
-          ? destination_node_info->namespace_()->str()
+      destination_node_info.namespace_()
```

Comment:
I don’t think this is used Or needed anymore. It can be removed if not needed.

Reply:
Thanks, done.